### PR TITLE
 plan: always clone `Expression` when set it to plan.

### PR DESCRIPTION
**Pull request by winoros**
Current master can be panicked by the test added.
PTAL @XuHuaiyu @zz-jason @lamxTyler 
Better be merged after #6260 and #6262

### CODE REVIEW

**Comment by zz-jason** [comment]
After #6260 and #6262 merged, we'd better evaluate the impact on sysbench of this patch.

File: plan/predicate_push_down.go [hunk 1]
```diff
@@ -133,8 +133,14 @@ func (p *LogicalJoin) PredicatePushDown(predicates []expression.Expression) (ret
 	case InnerJoin:
 		p.LeftConditions = nil
 		p.RightConditions = nil
-		p.EqualConditions = equalCond
-		p.OtherConditions = otherCond
+		p.EqualConditions = make([]*expression.ScalarFunction, 0, len(equalCond))
```
Comment (zhexuany) [inline plan/predicate_push_down.go]: `p.EqualConditions = make([]*expression.ScalarFunction, len(equalCond))` ? 

We need fulfill expression later anyway.

**Comment by XuHuaiyu** [comment]
LGTM

File: plan/predicate_push_down.go (hunk 1 above)
Comment (winoros) [reply plan/predicate_push_down.go]: Will this change affect too much?

Comment (coocood) [reply plan/predicate_push_down.go]: I think it's ok to use `append`.
We don't need to think if make full length is applicable, we can safely use append.

Comment (zhexuany) [reply plan/predicate_push_down.go]: I won't change as much. It is just neat to use the way I proposed.

Comment (zhexuany) [reply plan/predicate_push_down.go]: And yes. It's perfect fine to use append.

**Pull request merged**